app/src/main/java/com/example/login/presentation/LogInFragment.kt

```diff
@@ -0,0 +1,74 @@
+package com.example.login.presentation
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.activityViewModels
+import androidx.navigation.Navigation
+import com.example.login.viewmodel.UserViewModel
+import com.example.login.R
+import com.example.login.databinding.FragmentLogInBinding
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+class LogInFragment : Fragment() {
+    private lateinit var binding: FragmentLogInBinding
+    private val userViewModel: UserViewModel by activityViewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentLogInBinding.inflate(inflater, container, false)
+
+        binding.btnLogIn.setOnClickListener {
```

Comment:
onclick should be triggered inside viewmodel

Reply:
not done